Flagsmith.Client.Test/FlagsmithTest.cs

```diff
@@ -1,510 +1,544 @@
+            {
+                featuresDictionary.TryAdd($"Feature_{i}", 0);
+            }
+
+            // When 
+            var tasks = new Task[numberOfThreads];
+
+            // Create numberOfThreads threads.
+            for (int i = 0; i < numberOfThreads; i++)
+            {
+                var local = i;
+                string[] features = new string[callsPerThread];
+
+                // Prepare an array of feature names of length callsPerThread.
+                for (int j = 0; j < callsPerThread; j++)
+                {
+                    // The feature names are randomly selected from the featuresDictionary and added to the 
+                    // list of features, which represents the features that have been evaluated.  
+                    string featureName = $"Feature_{new Random().Next(1, featuresDictionary.Count + 1)}";
+                    features[j] = featureName;
+
+                    // The relevant key in the featuresDictionary is incremented to simulate an evaluation
+                    // to track for that feature. 
+                    featuresDictionary[featureName]++;
+                }
+
+                // Each thread will call IsFeatureEnabled for callsPerThread times.
+                tasks[i] = Task.Run(async () =>
+                {
+                    foreach (var feature in features)
+                    {
+                        await flags.IsFeatureEnabled(feature);
+                    }
+                });
+            }
+
+            await Task.WhenAll(tasks);
+
+            // Then
+            Dictionary<string, int> analyticsData = flagsmithClientTest.aggregatedAnalytics;
+            int totalCallsMade = 0;
+            foreach (var feature in featuresDictionary)
+            {
+                totalCallsMade += analyticsData[feature.Key];
+                Assert.Equal(feature.Value, analyticsData[feature.Key]);
+            }
+            Assert.Equal(numberOfThreads * callsPerThread, totalCallsMade);
+        }
+
+        [Fact]
+        public async Task TestGetIdentityFlagsTransientIdentityCallsExpected()
+        {
+            string identifier = "transient_identity";
+            bool transient = true;
+            var traits = new List<ITrait> { new Trait("some_trait", "some_value") };
+            var mockHttpClient = HttpMocker.MockHttpResponse(new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent(Fixtures.ApiTransientIdentityResponse)
+            });
+            var flagsmithClient = new FlagsmithClient(Fixtures.ApiKey, httpClient: mockHttpClient.Object);
+            var identityFlags = await flagsmithClient.GetIdentityFlags(identifier, traits, transient);
+            Assert.True(await identityFlags.IsFeatureEnabled("some_feature"));
+            Assert.Equal("some-identity-trait-value", await identityFlags.GetFeatureValue("some_feature"));

```

Comment:
I appreciate it's a bit more work since there's no implementation of `HttpMocker.verifyHttpRequest` that supports this, but I'd expect request body verification here.

Here we need to make sure the request is being sent with the `"transient": true` attribute present in the body.

Reply:
Okay, now we verify the body that is being sent.